ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -683,7 +717,177 @@ public void handleReplicaMetadataRequest(NetworkRequest request) throws IOExcept
+
+      // get the blob from the remote replica
+      ConnectedChannel connectedChannel =
+          connectionPool.checkOutConnection(remoteHostName, remoteDataNode.getPortToConnectTo(),
+              connectionPoolCheckoutTimeoutMs);
+      ChannelOutput channelOutput = connectedChannel.sendAndReceive(getRequest);
+      GetResponse getResponse = GetResponse.readFrom(channelOutput.getInputStream(), clusterMap);
+      if (getResponse.getError() != ServerErrorCode.No_Error) {
+        logger.error("ReplicateBlobRequest failed to get blob {} from the remote node {} {} {}", blobId, remoteHostName,
+            remoteHostPort, getResponse.getError());
+        return new Pair(getResponse.getError(), getResponse);
+      }
+      if ((getResponse.getPartitionResponseInfoList() == null) || (getResponse.getPartitionResponseInfoList().size()
+          == 0)) {
+        logger.error("ReplicateBlobRequest {} returned empty list from the remote node {} {} {}", blobId, remoteHostName,
+            remoteHostPort, getResponse.getError());
+        return new Pair(ServerErrorCode.Unknown_Error, getResponse);
+      }
+
+      // only have one partition. And checked at least it has one entry above.
+      PartitionResponseInfo partitionResponseInfo = getResponse.getPartitionResponseInfoList().get(0);
+      switch (partitionResponseInfo.getErrorCode()) {
+        case No_Error:
+          List<MessageInfo> messageInfoList = partitionResponseInfo.getMessageInfoList();
+          if (messageInfoList == null || messageInfoList.size() != 1) {
+            logger.error(
+                "ReplicateBlobRequest PartitionResponseInfo response from GetRequest {} {} {} {} {} returned null.",
+                partitionResponseInfo, messageInfoList, remoteHostName, remoteHostPort, blobId);
+            return new Pair(ServerErrorCode.Blob_Not_Found, getResponse);
+          } else {
+            return new Pair(ServerErrorCode.No_Error, getResponse);
+          }
+        case Blob_Deleted:
+          // Since GetOption is Include_All, even it's deleted on the remote replica, we'll still get the PutBlob.
+          // One exception is that because of compaction or other reasons, the PutRecord is gone.
+          logger.error("ReplicateBlobRequest Blob {} of {} is deleted on the source replica", blobId,
+              replicateBlobRequest);
+          return new Pair(ServerErrorCode.Blob_Deleted, getResponse);
+        case Blob_Authorization_Failure:
+          logger.error("ReplicateBlobRequest authorization failure for replicateBlobRequest {}", replicateBlobRequest);
+          return new Pair(ServerErrorCode.Blob_Authorization_Failure, getResponse);
+        default:
+          logger.error("ReplicateBlobRequest unknown failure {} for replicateBlobRequest {}",
+              partitionResponseInfo.getErrorCode(), replicateBlobRequest);
+          return new Pair(partitionResponseInfo.getErrorCode(), getResponse);
+      }
```

Comment:
This can be merged into just one statement. something like

``` 
if (partitionResponseInfo.getErrorCode() == No_Error) {
          xxxxxx
} else {
          logger.error("ReplicateBlobRequest failure {} for replicateBlobRequest {}",
              partitionResponseInfo.getErrorCode(), replicateBlobRequest);
          return new Pair(partitionResponseInfo.getErrorCode(), getResponse);
}
```

Reply:
I simplified it. 
But I'd like to keep the some comments. Hope it's ok.